src/blueprint/into_schema.rs

```diff
@@ -63,6 +63,58 @@ fn get_cache_key<'a, H: Hasher + Clone>(
   Some(key)
 }
 
+fn write_entity_cache<'a>(ctx: &'a EvaluationContext<'a, ResolverContext<'a>>, type_name: &str, output: &ConstValue) {
+  if let Some(Cache { max_age: ttl, hasher }) = ctx.req_ctx.type_cache_config.get(type_name) {
+    let hasher = hasher.clone();
+    if let Some(key) = get_cache_key(ctx, hasher) {
+      ctx.req_ctx.cache.insert(key, output.clone(), *ttl);
+    }
+  }
+
+  // if let Some(fld_caches) = ctx.req_ctx.field_cache_config.get(type_name) {
+  //   let fields_present_in_response = fld_caches
+  //     .iter()
+  //     .filter_map(|(k, c)| {
+  //       output.get_key(k).map(|r| (r, c))
+  //     });
+  //   for (field, Cache{max_age: ttl, hasher}) in fields_present_in_response {
+  //     let key = hasher.finish();
+  //     ctx.req_ctx.cache.insert(key, field.clone(), *ttl);
+  //   }
+  // }
+}
+
+async fn read_entity_cache<'a>(
+  ctx: &'a EvaluationContext<'a, ResolverContext<'a>>,
+  type_name: &str,
+) -> anyhow::Result<ConstValue> {
+  if let Some(Cache { hasher, .. }) = ctx.req_ctx.type_cache_config.get(type_name) {
+    let hasher = hasher.clone();
+
+    if let Some(key) = get_cache_key(ctx, hasher) {
+      return ctx
+        .req_ctx
+        .cache
+        .get(&key)
+        .ok_or(anyhow::anyhow!("NOT_PRESENT_IN_CACHE"));
+    }
+  }
+
+  Err(anyhow::anyhow!("NOT_PRESENT_IN_CACHE"))
+
+  // if let Some(fld_caches) = ctx.req_ctx.field_cache_config.get(type_name) {
+  //   let fields_present_in_response = fld_caches
+  //     .iter()
+  //     .filter_map(|(k, c)| {
+  //       output.get_key(k).map(|r| (r, c))
+  //     });
+  //   for (field, Cache{max_age: ttl, hasher}) in fields_present_in_response {
+  //     let key = hasher.finish();
+  //     ctx.req_ctx.cache.insert(key, field.clone(), *ttl);
+  //   }
+  // }
```

Comment:
do you need those commented code for anything?

Reply:
removed them. it wasn't required